src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -419,8 +419,14 @@ private void putNLPResultToSourceMapForMapType(
             for (Map.Entry<String, Object> inputNestedMapEntry : ((Map<String, Object>) sourceValue).entrySet()) {
                 if (sourceAndMetadataMap.get(processorKey) instanceof List) {
                     // build nlp output for list of nested objects
+                    List<Object> inputNestedMapValue = (List<Object>) inputNestedMapEntry.getValue();
+                    int valueIndex = 0;
```

Comment:
Why do we need `valueIndex`? You can simply iterate over the whole list

Reply:
Make sense, will change to iterator style.